### PR TITLE
Allow composition with .and

### PR DESCRIPTION
**Pull request by twhaples**
It will be helpful to be able to chain several matchers in a row on the same piece of code instead of running the piece of code multiple times.

### CODE REVIEW

File: rspec-que.gemspec [hunk 1]
```diff
@@ -1,11 +1,12 @@
 require File.expand_path('../lib/rspec/que/version', __FILE__)
+require 'date'
```
Comment (twhaples) [inline rspec-que.gemspec]: gemspec fails to compile without this on some versions of ruby

File: spec/rspec/que/queue_up_spec.rb [hunk 2]
```diff
@@ -160,5 +160,14 @@ class CJob; end
         end
       end
     end
+
+    describe '#and' do
+      it 'allows composition' do
+        expect { proc.call }.
+          to described_class.new(AJob).with('madoka').
```
Comment (twhaples) [inline spec/rspec/que/queue_up_spec.rb]: (`queue_up` in regular codes and `described_class.new` here are the same thing)

File: Gemfile.lock [hunk 3]
```diff
@@ -1,7 +1,7 @@
 PATH
   remote: .
   specs:
-    rspec-que (1.0.0)
+    rspec-que (1.0.1)
```
Comment (Baltoli) [inline Gemfile.lock]: I take it that this doesn't break what we already have, but only adds new features?

**Comment by Baltoli** [comment]
👍

File: Gemfile.lock (hunk 3 above)
Comment (twhaples) [inline Gemfile.lock]: that's the idea (just adds the `and` functionality)

**Pull request merged**